--- /usr/src/linux-source-2.6.32/fs/ext2/Makefile	2009-12-03 13:51:21.000000000 +1000
+++ Makefile	2010-10-13 09:41:06.716094619 +1000
@@ -1,13 +1,19 @@
 #
-# Makefile for the linux ext2-filesystem routines.
+# Makefile for the linux ext3301-filesystem routines.
 #
 
-obj-$(CONFIG_EXT2_FS) += ext2.o
+obj-m += ext3301.o
 
-ext2-y := balloc.o dir.o file.o ialloc.o inode.o \
-	  ioctl.o namei.o super.o symlink.o
+ext3301-y := acl.o balloc.o dir.o file.o ialloc.o inode.o \
+	  ioctl.o namei.o super.o symlink.o  \
+	  xattr.o xattr_user.o xattr_trusted.o xattr_security.o
 
-ext2-$(CONFIG_EXT2_FS_XATTR)	 += xattr.o xattr_user.o xattr_trusted.o
-ext2-$(CONFIG_EXT2_FS_POSIX_ACL) += acl.o
-ext2-$(CONFIG_EXT2_FS_SECURITY)	 += xattr_security.o
-ext2-$(CONFIG_EXT2_FS_XIP)	 += xip.o
+MOD_DIR=/lib/modules/$(shell uname -r)/build
+
+.PHONY: all
+all:
+	make -C $(MOD_DIR) M=$(PWD) modules
+
+.PHONY: clean
+clean:
+	make -C $(MOD_DIR) M=$(PWD) clean
